### PR TITLE
Add the ability to issue a reindex command to kinesis directly

### PR DESCRIPTION
**Pull request by jonathonherbert**
Adds a new reingest command. The docs describe it –

```
Reingest an image already present in the images bucket

USAGE
  $ grid image:reingest ID

ARGUMENTS
  ID  ID of image

OPTIONS
  -c, --compare          Do a dry run, and display the difference between the result and the data returned for this image from media-api
  -d, --dryRun           Display the result of the reingestion only
  -f, --force            Force reingestion if the image is already present in the Grid
  -h, --help             show CLI help
  -p, --profile=profile  [default: default] Profile name
```

### CODE REVIEW

**Comment by paperboyo** [comment]
Super-clear docs and useful too, thank you!

File: src/commands/image/reingest.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    } else {
+      this.log(JSON.stringify(projection, null, 2))
+    }
+  }
+
+
+  private async fetchProjection(id: string, adminToolsEndpoint: URL, reingest = true) {
+    let endpoint = `${adminToolsEndpoint}/${id}`
+    if (reingest) {
+      endpoint += '?reingest=true'
```
Comment (akash1810) [inline src/commands/image/reingest.ts]: minor: would be nice to use a `URLSearchParams` here.

File: src/commands/image/reingest.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        return
+      }
+      await this.wait(delayBetweenRequestsInMs)
+      currentIteration++
+    }
+    this.log('\n')
+    throw new Error(`Timed out trying to find image – tried ${maxIterations} times over ${maxIterations * delayBetweenRequestsInMs / 1000} seconds`)
+  }
+
+  private readonly wait = (delayInMs: number) => new Promise(resolve => setTimeout(resolve, delayInMs))
```
Comment (akash1810) [inline src/commands/image/reingest.ts:120]: nice

**Review by akash1810 — APPROVED** [review]
LGTM

File: src/commands/image/reingest.ts (hunk 1 above)
Comment (jonathonherbert) [reply src/commands/image/reingest.ts]: Yes! Good spot!

**Pull request merged**